bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/net/HttpRequest.java

```diff
@@ -0,0 +1,250 @@
+                return httpResponse;
+            } finally {
+                content.close();
+            }
+        } catch (ProcessingException | IllegalStateException | IOException e) {
+            return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, e.getMessage());
+        }
+    }
+
+    /**
+     * Send post command for a body comprised of XML to a URL with potentially some request headers
+     *
+     * @param url the non-null, non-empty URL
+     * @param body the non-null, possibly empty body (of XML)
+     * @param rqstHeaders the list of {@link Header} to add to the request
+     * @return the non-null http response
+     */
+    public HttpResponse sendPostXmlCommand(final String url, final String body, final Header... rqstHeaders) {
+        Validate.notEmpty(url, "url cannot be empty");
+        Objects.requireNonNull(body, "body cannot be null");
+
+        return sendPostCommand(url, body, MediaType.TEXT_XML + ";charset=utf-8", rqstHeaders);
+    }
+
+    /**
+     * Send post command for a body comprised of JSON to a URL with potentially some request headers
+     *
+     * @param url the non-null, non-empty URL
+     * @param body the non-null, possibly empty body (of JSON)
+     * @param rqstHeaders the list of {@link Header} to add to the request
+     * @return the non-null http response
+     */
+    public HttpResponse sendPostJsonCommand(final String url, final String body, final Header... rqstHeaders) {
+        Validate.notEmpty(url, "url cannot be empty");
+        Objects.requireNonNull(body, "body cannot be null");
+
+        return sendPostCommand(url, body, MediaType.APPLICATION_JSON, rqstHeaders);
+    }
+
+    /**
+     * Send post command to the specified URL with the body/media type and potentially some request headers
+     *
+     * @param url the non-null, non-empty URL
+     * @param body the non-null, possibly empty body (of JSON)
+     * @param mediaType the non-null, non-empty media type
+     * @param rqstHeaders the list of {@link Header} to add to the request
+     * @return the non-null http response
+     */
+    private HttpResponse sendPostCommand(final String url, final String body, final String mediaType,
+            final Header... rqstHeaders) {
+        Validate.notEmpty(url, "url cannot be empty");
+        Objects.requireNonNull(body, "body cannot be null");
+        Validate.notEmpty(mediaType, "mediaType cannot be empty");
+
+        try {
+            final Builder rqst = addHeaders(client.target(url).request(mediaType), rqstHeaders);
+            final Response content = rqst.post(Entity.entity(body, mediaType));
+
+            try {
+                final HttpResponse httpResponse = new HttpResponse(content);
+                return httpResponse;
+            } finally {
+                content.close();
+            }
+        } catch (IOException | ProcessingException e) {
+            return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, e.getMessage());
```

Comment:
would it be worth logging this error?

Reply:
Do you meant for the stack trace?  Not really - 99.99% of the time this simply means they are using the wireless interface and they turned off the device (which powers down the wireless port).  The getMessage() is the important part here - which will usually tell me exactly what I need